### PR TITLE
Add debug libfunc name when missing from debug info

### PR DESCRIPTION
**Pull request by edg-l**
This allows us to view libfunc names in MLIR with debug info when the sierra has no debug info, such as in contracts:

```mlir
    %12 = llvm.alloca %11 x !llvm.struct<(i64, array<24 x i8>)> {alignment = 8 : i64} : (i64) -> !llvm.ptr "enum_init(stmt_idx=10569)"
    %13 = llvm.alloca %11 x !llvm.struct<(i64, array<24 x i8>)> {alignment = 8 : i64} : (i64) -> !llvm.ptr "enum_init(stmt_idx=10555)"
    %14 = llvm.alloca %11 x !llvm.struct<(i64, array<24 x i8>)> {alignment = 8 : i64} : (i64) -> !llvm.ptr "enum_match(stmt_idx=10527)"
    %15 = llvm.alloca %11 x !llvm.struct<(i64, array<24 x i8>)> {alignment = 8 : i64} : (i64) -> !llvm.ptr "enum_init(stmt_idx=10542)"
    %16 = llvm.alloca %11 x !llvm.struct<(i64, array<24 x i8>)> {alignment = 8 : i64} : (i64) -> !llvm.ptr "enum_init(stmt_idx=10535)"
    %17 = llvm.alloca %11 x !llvm.struct<(i64, array<24 x i8>)> {alignment = 8 : i64} : (i64) -> !llvm.ptr "enum_init(stmt_idx=10481)"
    %18 = llvm.add %arg0, %11  : i64 "withdraw_gas(stmt_idx=10460)"
    %19 = llvm.icmp "uge" %arg1, %10 : i128 "withdraw_gas(stmt_idx=10460)"
    %20 = llvm.intr.usub.sat(%arg1, %10)  : (i128, i128) -> i128 "withdraw_gas(stmt_idx=10460)"
```

## Checklist
- [ ] Linked to Github Issue
- [ ] Unit tests added
- [ ] Integration tests added.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.

### CODE REVIEW

File: src/compiler.rs [hunk 1]
```diff
@@ -981,3 +1026,349 @@ where
...
+                "contract_address_try_from_felt252"
+            }
+            StarkNetConcreteLibfunc::ContractAddressToFelt252(_) => "contract_address_to_felt252",
+            StarkNetConcreteLibfunc::StorageRead(_) => "storage_read",
+            StarkNetConcreteLibfunc::StorageWrite(_) => "storage_write",
+            StarkNetConcreteLibfunc::StorageBaseAddressConst(_) => "storage_base_address_const",
+            StarkNetConcreteLibfunc::StorageBaseAddressFromFelt252(_) => {
+                "storage_base_address_from_felt252"
+            }
+            StarkNetConcreteLibfunc::StorageAddressFromBase(_) => "storage_address_from_bas",
```
Comment (JulianGCalderon) [inline src/compiler.rs:1327]:
```suggestion
            StarkNetConcreteLibfunc::StorageAddressFromBase(_) => "storage_address_from_base",
```

File: src/compiler.rs [hunk 2]
```diff
@@ -981,3 +1026,349 @@ where
...
+            StarkNetConcreteLibfunc::StorageAddressFromBaseAndOffset(_) => {
+                "storage_address_from_Base_and_offset"
```
Comment (JulianGCalderon) [inline src/compiler.rs:1329]:
```suggestion
                "storage_address_from_base_and_offset"
```

**Review by JulianGCalderon — COMMENTED** [review]
Also, I found a few typos

**Comment by edg-l** [comment]
this is added also by the other merged pr

**Pull request closed without merging**